### PR TITLE
[Bug fix] Fix DeepSeek slot remap for device sampling

### PR DESCRIPTION
**Pull request by peter941221**
### PR Category
Bug fix

### Summary
1. Fix DeepSeek vLLM decode sampling so `slot_remap` is converted into the row-padded seed-slot layout used by DeepSeek device sampling before the next decode-step seed advance.
2. Apply that converted remap before `seed_manager.get_new_values(...)` so condensed vLLM batches keep slot-to-seed state aligned across sampling rows.
3. Add focused host-side regressions for both boundaries:
   - the DeepSeek vLLM wrapper still forwards `slot_remap`
   - the DeepSeek sampling path expands logical user-slot remaps into the expected row-padded seed-slot remap before advancing seeds

### Notes for reviewers
1. The functional change is limited to the DeepSeek vLLM sampling bridge:
   - `models/demos/deepseek_v3/tt/generator_vllm.py`
   - `models/demos/deepseek_v3/tt/generator.py`
2. The key contract detail is that DeepSeek sampling uses row-padded seed slots:
   - logical user slots are mapped through `_sampling_device_slot(...)`
   - `SeedManager.apply_slot_remap(...)` expects remaps in that seed-slot index space, not in compact logical-user space
3. This follows the same seed-manager ordering already used in:
   - `models/tt_transformers/tt/generator.py`
   - `models/demos/llama3_70b_galaxy/tt/generator.py`
4. The issue here is not cache or decode math. It is slot-to-seed state drift after vLLM condenses the active batch and DeepSeek advances seeds against the wrong slot layout.
5. I kept the regression coverage host-side and focused on the contract boundary instead of trying to prove this through a full hardware-backed decode run.

### Test plan
1. `python3 -m py_compile models/demos/deepseek_v3/tt/generator.py`
2. `python3 -m py_compile models/demos/deepseek_v3/tt/generator_vllm.py`
3. `python3 -m py_compile models/demos/deepseek_v3/tests/test_sampling.py`
4. `python3 -m py_compile models/demos/deepseek_v3/tests/test_generator_vllm.py`
5. `git diff --check`
6. `python3` AST parse probe for the touched files
7. host-side logic probe confirming `_sampling_device_slot_remap(...)` expands `[7,6,5,4,3,2,1,0,15,14,13,12]` into the expected row-padded seed-slot remap
8. I could not complete local pytest collection in this WSL environment because the available Python envs here do not include `pytest`, and the available Python envs for this lane do not provide a ready importable TTNN runtime test environment.

### CODE REVIEW

**Comment by peter941221** [comment]
I rebased this onto current `main` and force-pushed the updated branch.

Current head: `4c9f515f7fa8c890f37f23ce15119580c9d63a04`.

The branch still covers the same bug family, but it now lands on the current DeepSeek sampling boundary:
- `decode_forward(...)` forwards `slot_remap` through `sample_decode_on_device(...)`
- `DeepseekGenerator` expands user-slot remaps into the row-padded seed-slot space before `seed_manager.get_new_values(...)`

I kept the regressions focused on those two boundaries:
- vLLM decode forwarding
- seed-slot remap ordering and row-padded mapping

File: models/demos/deepseek_v3/tests/test_generator_vllm.py [hunk 1]
```diff
@@ -44,6 +44,62 @@ def test_deepseek_generator_exposes_max_tokens_all_users_capability() -> None:
...
+    )
+
+    assert captured["super_decode_forward"]["tokens"].tolist() == [11, 22]
+    assert captured["super_decode_forward"]["sample_on_device"] is True
+    assert captured["sample_call"][0] == "decode_logits"
+    assert captured["sample_call"][1]["slot_remap"] == [7, 6, 5, 4]
+    assert fake_model._sampling_device_slot_remap(captured["sample_call"][1]["slot_remap"]) == [7, 6, 5, 4] + list(
+        range(8, 32)
+    ) + [39, 38, 37, 36] + list(range(40, 64))
+    assert captured["sample_call"][1]["skip_precompile"] is True
```
Comment (yieldthought) [inline models/demos/deepseek_v3/tests/test_generator_vllm.py]: Suggestion: keep this test focused on the vLLM pass-through boundary; the row-padded expansion is already covered in `test_sampling.py`.

```suggestion
    assert captured["sample_call"][1]["skip_precompile"] is True
```

**Comment by yieldthought** [comment]
Codex review:

[models/demos/deepseek_v3/tests/test_generator_vllm.py:96](https://github.com/tenstorrent/tt-metal/blob/4c9f515f7fa8c890f37f23ce15119580c9d63a04/models/demos/deepseek_v3/tests/test_generator_vllm.py#L96): The new host-side regression test will fail when run. _FakeModel._sampling_device_slot_remap calls DeepseekGenerator._sampling_device_slot_remap, and that helper now reads self.sampling_generator.seed_manager.max_batch_size, but _FakeModel never defines sampling_generator. Even if the fake had one, the expected remap here is for a two-row/12-entry remap, while the test passes only [7, 6, 5, 4]; the helper would return [7, 6, 5, 4] + list(range(4, 64)). Suggestion: remove this expansion assertion from the vLLM pass-through test, or add a proper fake seed manager and correct the expected value. The expansion path is already covered in test_sampling.py.

[models/demos/deepseek_v3/tt/generator_vllm.py:234](https://github.com/tenstorrent/tt-metal/blob/4c9f515f7fa8c890f37f23ce15119580c9d63a04/models/demos/deepseek_v3/tt/generator_vllm.py#L234): This comment now contradicts the implementation below: it says DeepSeek does not consume slot_remap and links the issue as still open, but [models/demos/deepseek_v3/tt/generator_vllm.py:262](https://github.com/tenstorrent/tt-metal/blob/4c9f515f7fa8c890f37f23ce15119580c9d63a04/models/demos/deepseek_v3/tt/generator_vllm.py#L262) forwards it into sample_decode_on_device. It also calls the incoming value a seed-slot remap even though generator.py converts it from compact user slots into row-padded seed-manager slots. Suggestion: update or remove the NOTE so future edits do not use the old contract.

**Comment by peter941221** [comment]
Addressed both review points.

I tightened `test_decode_forward_passes_slot_remap_to_device_sampling` so it only checks the vLLM pass-through boundary. The row-padded seed-slot expansion stays covered in `test_sampling.py`, and the fake model now provides the minimal `sample_decode_on_device(...)` hook that `decode_forward(...)` actually calls.

I also updated the stale `slot_remap` note in `generator_vllm.py` to match the current implementation. The vLLM bridge forwards compact user-slot remaps, and DeepSeek expands them inside the sampling path before seed advancement.

Local checks:
python3 -m py_compile models/demos/deepseek_v3/tt/generator_vllm.py models/demos/deepseek_v3/tests/test_generator_vllm.py models/demos/deepseek_v3/tests/test_sampling.py
git diff --check

I could not complete local pytest in this WSL environment because repo-level `conftest.py` imports `ttnn`, and the available TTNN runtime import currently fails on an existing MPI symbol boundary before these host-side tests start.
